extensions/aws/s3/S3WrapperBase.cpp

```diff
@@ -32,6 +33,17 @@ namespace minifi {
 namespace aws {
 namespace s3 {
 
+void GetObjectResult::setFilePaths(const std::string& key) {
+  const auto last_slash = key.find_last_of('/');
+  absolute_path = key;
+  if (last_slash != std::string::npos) {
+    path = key.substr(0, last_slash);
+    filename = key.substr(last_slash + 1);
+  } else {
+    filename = key;
```

Comment:
Don't we have a splitter in pathutils or stringutils already?

Reply:
You are right there is a getFileNameAndPath in pathutils, but that uses OS specific separators, but in AWS the object key always uses '/' character as a separator.

Reply:
I think some of our such util functions already have a "force posix" bool argument to address this, I think we can apply that for this specific function and use it here.

Reply:
Found it in the FileUtils, the split_path function does what we need here, thanks! Added it in [fd2e4fa](https://github.com/apache/nifi-minifi-cpp/pull/970/commits/fd2e4facc77e51e6e6fe05280c9ab6d563ee5114)